public/app/features/migrate-to-cloud/onprem/Page.tsx

```diff
@@ -1,32 +1,69 @@
 import { skipToken } from '@reduxjs/toolkit/query/react';
 import React, { useState } from 'react';
 
-import { Button, Stack, Text } from '@grafana/ui';
-import { Trans } from 'app/core/internationalization';
+import { Alert, Box, Button, Stack } from '@grafana/ui';
+import { Trans, t } from 'app/core/internationalization';
 
-import { useGetStatusQuery, useListResourcesQuery } from '../api';
+import { useGetStatusQuery, useListResourcesQuery, useStartMigrationMutation } from '../api';
 
 import { DisconnectModal } from './DisconnectModal';
 import { EmptyState } from './EmptyState/EmptyState';
+import { MigrationInfo } from './MigrationInfo';
 import { ResourcesTable } from './ResourcesTable';
 
 export const Page = () => {
   const { data: status, isFetching } = useGetStatusQuery();
   const { data: resources } = useListResourcesQuery(status?.enabled ? undefined : skipToken);
+  const [startMigration, { isLoading: startMigrationIsLoading, isError: startMigrationIsError }] =
+    useStartMigrationMutation();
   const [isDisconnecting, setIsDisconnecting] = useState(false);
 
   if (!status?.enabled) {
     return <EmptyState />;
   }
 
+  const isBusy = isFetching || isDisconnecting || startMigrationIsLoading;
+
   return (
     <>
-      <Stack direction="column" alignItems="flex-start">
-        {status.stackURL && <Text variant="h4">{status.stackURL}</Text>}
+      <Stack direction="column" gap={4}>
+        {startMigrationIsError && (
+          <Alert
+            severity="error"
+            title={t(
+              'migrate-to-cloud.summary.error-starting-migration',
+              'There was an error starting cloud migration'
+            )}
+          />
+        )}
+
+        {status.stackURL && (
+          <Box borderColor="weak" borderStyle="solid" padding={2}>
+            <Stack gap={4} alignItems="center">
+              <MigrationInfo
+                title={t('migrate-to-cloud.summary.target-stack-title', 'Uploading to')}
+                value={
+                  <>
+                    {status.stackURL}{' '}
+                    <Button onClick={() => setIsDisconnecting(true)} disabled={isBusy} variant="secondary" size="sm">
+                      <Trans i18nKey="migrate-to-cloud.summary.disconnect">Disconnect</Trans>
+                    </Button>
+                  </>
+                }
+              />
+
+              <div style={{ flex: '1 1 auto' }} />
```

Comment:
minor nit: think `justifyContent: 'space-between'` is better, it's what it's made for 😄

Reply:
Yeah, I was lazy here...


Problem is the actual layout isn't space-between 

![image](https://github.com/grafana/grafana/assets/46142/d8f0309b-f13b-45ac-959f-4cca0bbd9dfa)

Reply:
is that not `space-between`? you'd have something like:
```
<Stack space-between>
  <Stack gap={4}>
    <MigrationInfo />
    <MigrationInfo />
    <MigrationInfo />
    <MigrationInfo />
  </Stack>
  <Button>Upload everything</Button>
</Stack>
```

Reply:
ahh yeah, you're right 👍